examples/EllipticalSliceSampling.jl

```diff
@@ -0,0 +1,118 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,jl:light
+#     text_representation:
+#       extension: .jl
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.4.2
+#   kernelspec:
+#     display_name: Julia 1.4.1
+#     language: julia
+#     name: julia-1.4
+# ---
+
+# # An example showcasing the usage of EllipticalSliceSampling.jl with AbstractGPs.jl
+
+# Loading the necessary packages.
+
+using AbstractGPs, KernelFunctions, Plots, CSV, DataFrames
+include("utils.jl")
+
+# Loading [toy regression dataset](https://github.com/GPflow/docs/blob/master/doc/source/notebooks/basics/data/regression_1D.csv) taken from GPFlow examples.
+
+df = CSV.read("data/regression_1D.csv", header=false);
+x = df[:, 1];
+y = df[:, 2];
+
+# Making a custom kernel with two parameters.
+
+k = ScaledKernel(transform(Matern52Kernel(), ScaleTransform(1.0)), 1.0)
+
+# Instantiating a Gaussian Process with the given kernel `k`.
+
+f = GP(k)
+
+# Instantiating a `FiniteGP`, a finite dimentional projection at the inputs of the dataset observed under Gaussian Noise with $\sigma = 0.001$ .
+
+fx = f(x, 0.001)
+
+# Data's log-likelihood w.r.t prior `GP`. 
+
+logpdf(fx, y)
+
+# Calculating the exact posterior with the given expected output `y` and `FiniteGP`. THe GP's kernel currently has fixed parameters. 
+
+p_fx = posterior(fx, y)
+
+# Data's log-likelihood w.r.t exact posterior `GP`. We see that it drastically increases.
+
+logpdf(p_fx(x), y)
+
+# Plotting the functions sampled from the exact posterior `p_fx` along with the data points.
+
+plt = scatter(x, y, label = "Data")
+sampleplot!(plt, p_fx(sort(x), 0.001), 100, alph=0.1)
+
+# # Elliptical Slice Sampler
+
+# Previously, we computed the the exact posterior GP without tuning the kernel parameters and achieved a loglikelihood on exact posterior of $-1.285$. We now attempt get a better posterior by sampling for kernel parameters using Elliptical Slice Sampler provided by [EllipticalSliceSampling.jl](https://github.com/TuringLang/EllipticalSliceSampling.jl/)
+
+# We start of by loading necessary packages.
+
+using EllipticalSliceSampling, Distributions
+
+# We define a function which returns log-likelihood of of data w.r.t an exact posterior with given set of kernel parameters.
+
+function logp(params)
+    exp_params = exp.(params)
+    kernel = ScaledKernel(transform(Matern52Kernel(), ScaleTransform(exp_params[1])), exp_params[2])
+    f = GP(kernel)
+    fx = f(x, 0.1)
+    p_fx = posterior(fx, y)
+    return logpdf(p_fx(x, 0.1), y)
+end
+
+# We define a Gaussian prior over the joint distribution on kernel parameters space. Since we have only two parameters, we define a multi-variate Gaussian of dimension two.
+
+prior = MvNormal(zeros(2), ones(2) .* 1)
```

Comment:
```suggestion
prior = MvNormal(2, 1)
```
(see https://github.com/JuliaStats/Distributions.jl/blob/feae3ba9ada360d2980c901e7c737ca02bda7a31/src/multivariate/mvnormal.jl#L221). Another alternative would be
```suggestion
prior = MvNormal(ones(2))
```